src/utils/app.context.js

```diff
@@ -0,0 +1,169 @@
+/* eslint-disable react/sort-comp */
+import { createContext, Component } from 'preact';
+import { useContext } from 'preact/hooks';
+import uhttpdService from './uhttpd.service';
+
+const UNAUTH_SESSION_ID = '00000000000000000000000000000000';
+const DEFAULT_TIMEOUT = 5000;
+const DEFAULT_COMMUNITY_SETTINGS = {
+	bad_signal: '-82',
+	acceptable_loss: '20',
+	bad_bandwidth: '1',
+	good_signal: '-65',
+	good_bandwidth: '5'
+};
+
+/** Context used to pass application wise data and services to nested components */
+export const AppContext = createContext();
+
+class DisconnectedError extends Error {}
+
+export class AppContextProvider extends Component {
+
+	constructor(props) {
```

Comment:
Would be good to know what explicit attributes could be passed to props... can be documented as a jsdoc for example